### PR TITLE
feat: New spec and parser to get the expire date of ssl certificate in rsyslog

### PR DESCRIPTION
**Pull request by huali027**
* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### CODE REVIEW

**Comment by huali027** [comment]
@xiangce @JoySnow Please help to review, thanks.

File: insights/specs/datasources/ssl_certificate.py [hunk 1]
```diff
@@ -102,3 +104,31 @@ def mssql_tls_cert_file(broker):
...
+    rsyslog_objs = broker[RsyslogAllConf]
+    for obj in rsyslog_objs.values():
+        for item in obj:
+            if 'DefaultNetstreamDriverCertFile' in item:
+                # basic format
+                if '$DefaultNetstreamDriverCertFile' in item:
+                    return item.split()[-1].strip()
+                else:
+                    # advanced format, it is set in global block, and the global line contains all the content in it
+                    parts = item.strip().split()
```
Comment (JoySnow) [inline insights/specs/datasources/ssl_certificate.py:130]: From the parser `RsyslogConf`, all lines from it had been `strip()`. Seems no need to have one more round of `strip()` here?

File: insights/specs/datasources/ssl_certificate.py [hunk 2]
```diff
@@ -102,3 +104,31 @@ def mssql_tls_cert_file(broker):
...
+    """
+    rsyslog_objs = broker[RsyslogAllConf]
+    for obj in rsyslog_objs.values():
+        for item in obj:
+            if 'DefaultNetstreamDriverCertFile' in item:
+                # basic format
+                if '$DefaultNetstreamDriverCertFile' in item:
+                    return item.split()[-1].strip()
+                else:
+                    # advanced format, it is set in global block, and the global line contains all the content in it
```
Comment (JoySnow) [inline insights/specs/datasources/ssl_certificate.py:129]: IIUC, this `if` and `else` are describing two parallel cases: `basic format` or `advanced format` . If this is the case, it is better to have the two comment lines with the same indent.

Comment (JoySnow) [inline insights/specs/datasources/ssl_certificate.py:129]: BTW, just noticed that this line and also line#24 are pretty long in this file's view. Would you like help to "prettify" the lines' length in this file? Say, shorten to around length 80 ...

**Review by JoySnow — CHANGES_REQUESTED** [review]

File: insights/specs/default.py [hunk 3]
```diff
@@ -540,6 +540,7 @@ class DefaultSpecs(Specs):
     rpm_ostree_status = simple_command("/usr/bin/rpm-ostree status --json", signum=signal.SIGTERM)
     rpm_pkgs = rpm_pkgs.pkgs_with_writable_dirs
     rsyslog_conf = glob_file(["/etc/rsyslog.conf", "/etc/rsyslog.d/*.conf"])
+    rsyslog_tls_cert_enddate = command_with_args("/usr/bin/openssl x509 -in %s -enddate -noout", ssl_certificate.rsyslog_tls_cert_file)
```
Comment (JoySnow) [inline insights/specs/default.py:543]: Do we need to also add this spec to file `insights/specs/insights_archive.py` ?    @xiangce @huali027

Comment (huali027) [reply insights/specs/default.py:543]: @JoySnow I thought `command_with_args ` is just used by "core-collect", so no need to add it. @xiangce Please help do the final confirmation.

File: insights/specs/datasources/ssl_certificate.py (hunk 2 above)
Comment (huali027) [reply insights/specs/datasources/ssl_certificate.py]: @JoySnow I adjust it to make it shorter, please help to confirm. 
We can define the standard about the max length of code and comment we must follow, then we can pay attention to it later.

**Comment by huali027** [comment]
@JoySnow For other comment, I have updated the code, please help to review again, thanks.

File: insights/specs/default.py (hunk 3 above)
Comment (xiangce) [reply insights/specs/default.py:543]: The `insights_archives.py` is a way (or workaround) for the new core collection to keep compatible with specs that is collected by the legacy collection (aka `uploader.json`)

For new specs that was not added to the `uploader.json`, like `command_with_args` kind specs, it's not required to add its `mangle_name` to the `insights_archives.py`

Comment (JoySnow) [reply insights/specs/default.py:543]: Thanks for the info and confirmation, @huali027, @xiangce .

**Review by JoySnow — APPROVED** [review]
Thanks for the updates.  LGTM now.

**Pull request merged**